### ISSUE TITLE
activation function

### ISSUE THREAD

**Issue by happyzhichengzhao** [philipperemy/keras-tcn#55]
Somewhere you seem to have forgotten to use the activation function

**Comment by philipperemy** [comment]
@happyzhichengzhao can you be a bit more precise here?

**Comment by happyzhichengzhao** [comment]
> @happyzhichengzhao can you be a bit more precise here?

`    # 1x1 conv to match the shapes (channel dimension).
    prev_x = Conv1D(nb_filters, 1, padding='same')(prev_x)
    res_x = keras.layers.add([prev_x, x])`

should be replaced by

`# 1x1 conv to match the shapes (channel dimension).
    if prev_x.shape[2] != nb_filters:
        prev_x = Conv1D(nb_filters, 1, padding='same')(prev_x)
    res_x = keras.layers.add([prev_x, x])
    res_x = Activation('relu')(res_x)
    return res_x, x`

**Comment by happyzhichengzhao** [comment]
> @happyzhichengzhao can you be a bit more precise here?
in line 110

        x = inputs
        # 1D FCN.
        x = Convolution1D(self.nb_filters, 1, padding=self.padding)(x)

Does it also need an activation function here?

**Comment by philipperemy** [comment]
@happyzhichengzhao what does the paper say?

https://arxiv.org/pdf/1803.01271.pdf

I'll have a look next week but if you can already point out, it would be helpful!

**Comment by philipperemy** [comment]
https://github.com/philipperemy/keras-tcn/pull/57

**Comment by philipperemy** [comment]
Fixed in 2.7.0
